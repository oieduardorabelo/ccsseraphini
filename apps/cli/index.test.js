jest.mock('open', () => (input) => input);

let { main } = require('./index');

describe('cc @sseraphini / cli', () => {
  let logOriginal;
  let logStub = jest.fn();
  beforeEach(() => {
    logOriginal = console.log;
    console.log = logStub;
  });

  afterEach(() => {
    console.log = logOriginal;
    process.argv = [];
    logStub.mockClear();
  });

  it('show help', async () => {
    process.argv = [null, null];
    await main();
    expect(logStub.mock.calls[0][0]).toMatchSnapshot();

    process.argv = [null, null, '-h'];
    await main();
    expect(logStub.mock.calls[0][0]).toMatchSnapshot();

    process.argv = [null, null, '--help'];
    await main();
    expect(logStub.mock.calls[0][0]).toMatchSnapshot();
  });

  it('show version', async () => {
    process.argv = [null, null, '-v'];
    await main();
    expect(logStub.mock.calls[0]).toMatchSnapshot();

    process.argv = [null, null, '--version'];
    await main();
    expect(logStub.mock.calls[0]).toMatchSnapshot();
  });

  it('show error', async () => {
    process.argv = [null, null, '-blah'];
    await main();
    expect(logStub.mock.calls[0]).toMatchSnapshot();
  });

  it('opens url with text', async () => {
    process.argv = [null, null, 'how do i tweet'];
    let url = await main();
    expect(url).toMatchInlineSnapshot(
      `"https://twitter.com/intent/tweet?text=how do i tweet%0Acc%20%40sseraphini"`,
    );
  });
});
