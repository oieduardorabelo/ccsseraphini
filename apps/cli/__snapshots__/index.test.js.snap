// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cc @sseraphini / cli show error 1`] = `
Array [
  "[31mError[0m: Found argument '[31m-blah[0m' which wasn't expected, or isn't valid in this context!

[33mUSAGE[0m:
  npx ccsseraphini [QUESTION]

For more information try [32m--help[0m or [32m--h[0m",
]
`;

exports[`cc @sseraphini / cli show help 1`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE:[0m
  npx ccsseraphini [QUESTION]

[33mEXAMPLE:[0m
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS:[0m
  [32m-h[0m, [32m--help[0m      Print this help message.
  [32m-v[0m, [32m--version[0m   Show version information."
`;

exports[`cc @sseraphini / cli show help 2`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE:[0m
  npx ccsseraphini [QUESTION]

[33mEXAMPLE:[0m
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS:[0m
  [32m-h[0m, [32m--help[0m      Print this help message.
  [32m-v[0m, [32m--version[0m   Show version information."
`;

exports[`cc @sseraphini / cli show help 3`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE:[0m
  npx ccsseraphini [QUESTION]

[33mEXAMPLE:[0m
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS:[0m
  [32m-h[0m, [32m--help[0m      Print this help message.
  [32m-v[0m, [32m--version[0m   Show version information."
`;

exports[`cc @sseraphini / cli show version 1`] = `
Array [
  "[32mcli 0.0.1[0m",
]
`;

exports[`cc @sseraphini / cli show version 2`] = `
Array [
  "[32mcli 0.0.1[0m",
]
`;
